staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -596,21 +609,33 @@ func (s *store) List(ctx context.Context, key, resourceVersion string, pred stor
 			growSlice(v, 2048, len(getResp.Kvs))
 		}
 
-		// take items from the response until the bucket is full, filtering as we go
-		for _, kv := range getResp.Kvs {
```

Comment:
I feel like we should have some helper for parallelizing a `for` loop.  Inlining a bunch of channel / wait group / workers logic into our already complicated storage layer does not sound like a good idea.

Reply:
I was able to simplify this quite a bit, PTAL.